gordon_janitor_gcp/gce_clients.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Client classes to retrieve project and instance data from GCE.
+
+These clients use the asynchronous HTTP client defined in
+:py:mod:`gordon_janitor_gcp.http_client` and require service account or
+JWT-token credentials for authentication.
+
+To use:
+
+.. code-block:: python
+
+    import asyncio
+
+    import aiohttp
+
+    from gordon_janitor_gcp import auth
+
+    loop = asyncio.get_event_loop()
+
+    async def main():
+        session = aiohttp.ClientSession()
+        loop = asyncio.get_event_loop()
+        auth_client = auth.GoogleAuthClient(k
+            eyfile='/path/to/keyfile', session=session)
+        client = GCPClient(auth_client, session)
+        instances = await client.list_instances('project-id')
+        print(instances)
+
+    loop.run_until_complete(main())
+    # example output
+    # [{'hostname': 'instance-1', 'internal_ip': '10.10.10.10',
+    #   'external_ip': '192.168.1.10'}]
+"""
+
+import logging
+
+from gordon_janitor_gcp import http_client
+
+
+class CRMClient(http_client.AIOGoogleHTTPClient,
```

Comment:
This is more of a nitpick, but I'm wondering if we should standardize on naming. There's `GDNSClient`\*, `GCPClient` (though should be renamed to `GCEClient`), and then `CRMClient` - maybe `GCRMClient` and something like `GPaginatorMixin`, or something else?

\* FTR I have a local WIP patch that renames `GoogleDNSReconciler` to `GDNSReconciler`, `GoogleAuthClient` to `GAuthClient`, `AIOGoogleDNSClient` to `GDNSClient`, `AIOGoogleHTTPClient` to `AIOConnection`, and `GooglePubsubPublisher` (still the open PR) to `GPubsubPublisher` (which is both the publisher interface and API client of pub/sub).

Thoughts?